src/services/positionService.ts

```diff
@@ -742,6 +834,35 @@ export default class PositionService {
     return this.safeService.submitMultipleTxs([approveTx, depositTx]);
   }
 
+  async xCallDeposit(
```

Comment:
I would prefer if we keep it all under the `deposit` function, the consumers of the positionService should have no need to know if it is a normal deposit or an xCallDeposit, and we can query different functions based on the destination network.

Reply:
bump, from what I've said before about accepting a `fundWithToken` that is of `Token` type. If the `from.chainId !== fundWithToken.chainId` then we go through xCall, if not, we go through the normal flow